pkg/daemon/ceph/client/osd.go

```diff
@@ -220,6 +221,58 @@ func GetOSDUsage(context *clusterd.Context, clusterInfo *ClusterInfo) (*OSDUsage
 	return &osdUsage, nil
 }
 
+func ResizeCrushTreeBucketWeight(atleastOneOsdGotResize bool, ctx *clusterd.Context, clusterInfo *ClusterInfo) error {
+	if !atleastOneOsdGotResize {
+		return nil
+	}
+	logger.Info("update crush tree bucket weight, as osd are resized for cluster in namespace %q", clusterInfo.Namespace)
+	args := []string{"osd", "crush", "reweight-all"}
+	buf, err := NewCephCommand(ctx, clusterInfo, args).Run()
+	if err != nil {
+		return errors.Wrapf(err, "failed to reweight crush tree bucket weight for cluster in namespace %q: %s", clusterInfo.Namespace, string(buf))
+	}
+	return nil
+}
+
+func convertKibibytesToTebibytes(kib string) (float64, error) {
+	kibFloat, err := strconv.ParseFloat(kib, 64)
+	if err != nil {
+		return float64(0), errors.Wrap(err, "failed to convert string to float")
+	}
+	return kibFloat / float64(1024*1024*1024), nil
+}
+
+func ResizeOsdCrushWeight(actualOSD OSDNodeUsage, ctx *clusterd.Context, clusterInfo *ClusterInfo) (bool, error) {
+	crushWeight, err := strconv.ParseFloat(actualOSD.CrushWeight.String(), 64)
+	if err != nil {
+		return false, errors.Wrapf(err, "failed converting string to float for osd.%d crush weight %q", actualOSD.ID, actualOSD.CrushWeight.String())
+	}
+	// actualOSD.KB is in KiB units
+	desiredOSDTiB, err := convertKibibytesToTebibytes(actualOSD.KB.String())
+	if err != nil {
+		return false, errors.Wrapf(err, "failed to convert KiB to TiB for osd.%d crush weight %q", actualOSD.ID, actualOSD.KB.String())
+	}
+
+	// do not reweight if the desired crush weight is 0 or they are equal or there percentage resize is less than 10 percent
+	if desiredOSDTiB == float64(0) {
+		logger.Debugf("desired osd size is 0 for osd.%d, not resizing the crush weights", actualOSD.ID)
+		return false, nil
+	} else if math.Abs(((desiredOSDTiB - crushWeight) / crushWeight)) < 0.1 {
```

Comment:
Need unit tests for this method, especially to check the calculations such as this

Reply:
OKay, on upper level we are checking but I will try to check with differnt osd size inputs

Reply:
The current desired osd values, actually clarifies it,
And I have also done testing for this locally,

Reply:
If you think of any specific test, other than this  I am willing to add